File: spring-dao/src/main/java/org/r3al/springdao/annotations/DaoQueryParam.java
package org.r3al.springdao.annotations;

import org.r3al.springdao.DaoQueryActionType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.PARAMETER, ElementType.FIELD, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface DaoQueryParam {
    String value();
    boolean addChildren() default false;
    DaoQueryActionType action() default DaoQueryActionType.DEFAULT;
}

File: spring-dao/src/main/kotlin/org/r3al/springdao/annotations/DaoQueryParam.kt
package org.r3al.springdao.annotations

import org.r3al.springdao.DaoQueryOperator
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Target(
    AnnotationTarget.VALUE_PARAMETER,
    AnnotationTarget.FIELD,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Retention(
    RetentionPolicy.RUNTIME
)
annotation class DaoQueryParam(
    val value: String,
    val operator: DaoQueryOperator = DaoQueryOperator.DEFAULT,
    val addChildren: Boolean = false
)